src/shell/components/FieldTypeUUID/FieldTypeUUID.js

```diff
@@ -18,39 +23,78 @@ export const FieldTypeUUID = React.memo(function FieldTypeUUID(props) {
     }
   }, []);
 
+  const handleCopyClick = (value) => {
+    navigator.clipboard
+      ?.writeText(value)
+      .then(() => {
+        setIsCopied(true);
+        setTimeout(() => {
+          setIsCopied(false);
+        }, 5000);
+      })
+      .catch((err) => {
+        // Handle errors
+        console.error(err);
+      });
+  };
   return (
     <label className={cx(styles.FieldTypeUUID, props.className)}>
-      <div className={styles.DateFieldTypeInput}>
-        <FontAwesomeIcon
-          className={styles.Icon}
-          icon={faClipboard}
-          aria-hidden="true"
-          title="Click to Copy"
-          onClick={(e) => {
-            const input = document.createElement("input");
-            document.body.appendChild(input);
-            input.value = props.value;
-            input.focus();
-            input.select();
-            const result = document.execCommand("copy");
-            input.remove();
-            if (result === "unsuccessful") {
-              return props.dispatch(
-                notify({
-                  type: "error",
-                  message: "Failed to copy the team ID to your clipboard",
-                })
-              );
-            }
+      <div>
+        <Tooltip
+          slotProps={{
+            popper: {
+              modifiers: [
+                {
+                  name: "offset",
+                  options: {
+                    offset: [0, -47],
+                  },
+                },
+              ],
+            },
           }}
-        />
+          placement="top"
+          title="This field cannot be edited"
+        >
+          <TextField
+            required={props.required}
+            value={props.value || ""}
+            fullWidth
+            type="text"
+            /**
+             * Readonly Props override the default color for the inputs
+             * This reset the color to the default text color
+             */
+            sx={{
+              ".MuiInputBase-readOnly": {
+                color: "text.primary",
+              },
+            }}
+            InputProps={{
+              readOnly: true,
```

Comment:
@agalin920 

I added readonly prop instead, but this override the default font color, so I had to target the readonly classes to reset it back to text.primary.